`build-static.sh` produces unusable `mold-wrapper.so`

The build script sets its own `LDFLAGS` for use inside the container. Since these `LDFLAGS` contain `-static`, the linker tries to produce `mold-wrapper.so` as a **statically** linked **dynamic** library - which is sort of an oxymoron. As a result, it is not possible to `LD_PRELOAD` the library.

How to reproduce:
```sh
make clean
./build-static.sh
make test
```

Relevant part of the output:
```
Testing mold-wrapper2 ...       not a dynamic executable
ERROR: ld.so: object '/home/christoph/Projects/mold/out/test/elf/mold-wrapper2/bin/mold-wrapper.so' from LD_PRELOAD cannot be preloaded (cannot dynamically load executable): ignored.
OK
Testing mold-wrapper ...        not a dynamic executable
ERROR: ld.so: object '/home/christoph/Projects/mold/test/elf/../../mold-wrapper.so' from LD_PRELOAD cannot be preloaded (cannot dynamically load executable): ignored.
ERROR: ld.so: object '/home/christoph/Projects/mold/test/elf/../../mold-wrapper.so' from LD_PRELOAD cannot be preloaded (cannot dynamically load executable): ignored.
/usr/bin/ld: cannot find execl: No such file or directory
make[1]: *** [Makefile.linux:6: elf/mold-wrapper.sh] Error 1
make: *** [Makefile:145: test] Error 2
```

Suggested solution: make sure that the custom `LDFLAGS` are applied only to `mold` itself, not to `mold-wrapper.so`.

--- a/Makefile
+++ b/Makefile
@@ -105,7 +105,7 @@ endif
 all: mold mold-wrapper.so
 
 mold: $(OBJS) $(MIMALLOC_LIB) $(TBB_LIB) $(XXHASH_LIB)
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $(OBJS) -o $@ $(LIBS)
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(OBJS) -o $@ $(LIBS)
 	ln -sf mold ld
 	ln -sf mold ld64.mold
 

--- a/build-static.sh
+++ b/build-static.sh
@@ -23,16 +23,16 @@ RUN apt-get update && \
   rm -rf /var/lib/apt/lists/*
 EOF
 
-LDFLAGS='-fuse-ld=lld -static'
+EXTRA_LDFLAGS='-fuse-ld=lld -static'
 
 # libstdc++'s `std::__glibcxx_rwlock_rdlock` refers these symbols
 # as weak symbols, although they need to be defined. Otherwise,
 # the program crashes after juping to address 0.
 # So, we force loading symbols as a workaround.
-LDFLAGS="$LDFLAGS -Wl,-u,pthread_rwlock_rdlock"
-LDFLAGS="$LDFLAGS -Wl,-u,pthread_rwlock_unlock"
-LDFLAGS="$LDFLAGS -Wl,-u,pthread_rwlock_wrlock"
+EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-u,pthread_rwlock_rdlock"
+EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-u,pthread_rwlock_unlock"
+EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-u,pthread_rwlock_wrlock"
 
 docker run -it --rm -v "`pwd`:/mold" -u $(id -u):$(id -g) \
   mold-build-ubuntu20 \
-  make -C /mold -j$(nproc) LDFLAGS="$LDFLAGS"
+  make -C /mold -j$(nproc) EXTRA_LDFLAGS="$EXTRA_LDFLAGS"

